sw/device/silicon_creator/lib/manifest.h

```diff
@@ -0,0 +1,210 @@
+static_assert(offsetof(manifest_t, signature) == 8, "");
+static_assert(offsetof(manifest_t, image_length) == 392, "");
+static_assert(offsetof(manifest_t, image_version) == 396, "");
+static_assert(offsetof(manifest_t, image_timestamp) == 400, "");
+static_assert(offsetof(manifest_t, exponent) == 408, "");
+static_assert(offsetof(manifest_t, reserved1) == 412, "");
+static_assert(offsetof(manifest_t, usage_constraints) == 416u, "");
+static_assert(offsetof(manifest_t, lockdown_info) == 448u, "");
+static_assert(offsetof(manifest_t, modulus) == 464u, "");
+static_assert(offsetof(manifest_t, extension0_offset) == 848u, "");
+static_assert(offsetof(manifest_t, extension0_checksum) == 852u, "");
+static_assert(offsetof(manifest_t, extension1_offset) == 856u, "");
+static_assert(offsetof(manifest_t, extension1_checksum) == 860u, "");
+static_assert(offsetof(manifest_t, extension2_offset) == 864u, "");
+static_assert(offsetof(manifest_t, extension2_checksum) == 868u, "");
+static_assert(offsetof(manifest_t, extension3_offset) == 872u, "");
+static_assert(offsetof(manifest_t, extension3_checksum) == 876u, "");
+static_assert(sizeof(manifest_t) == 880, "");
+
+/**
+ * Helper macro for defining the manifest of a boot stage stored in flash.
+ *
+ * This definition places the manifest struct in the `.manifest` section, which
+ * is the first section of boot stages stored in flash, e.g. see
+ * sw/device/silicon_creator/rom_exts/rom_ext_common.ld. The `used` attribute
+ * ensures that the compiler emits the definition even if it is not used.
+ *
+ * Boot stages should not need to access their own manifests but since the
+ * manifest is updated using the signer tool, the manifest struct is
+ * volatile-qualified in case this assumption changes in the future.
+ */
+#define DEFINE_FLASH_BOOT_STAGE_MANIFEST(name)                                \
+  static const volatile manifest_t name __attribute__((section(".manifest"))) \
+      __attribute__((used))
+
+/**
+ * Signed region of an image.
+ */
+typedef struct manifest_signed_region {
+  /**
+   * Start of the signed region of an image.
+   */
+  const void *start;
+  /**
+   * Length of the signed region of an image in bytes.
+   */
+  size_t length;
+} manifest_signed_region_t;
+
+/**
+ * Gets the signed region of an image.
+ *
+ * This function also performs a basic check to ensure that the length of the
+ * image is within a reasonable range.
+ *
+ * @param manifest A manifest
+ * @param[out] signed_region Signed region of an image.
+ * @return The result of the operation.
+ */
+inline rom_error_t manifest_signed_region_get(
+    const manifest_t *manifest, manifest_signed_region_t *signed_region) {
+  // FIXME: We should also have a lower bound.
+  enum { kMaxLength = 64 * 1024 };
+  if (manifest->image_length > kMaxLength) {
+    return kErrorManifestInternal;
+  }
+  *signed_region = (manifest_signed_region_t){
+      .start = (const void *)&manifest->image_length,
```

Comment:
IIRC you never need to explicitly cast to void.

Reply:
Yes, removed.